json_serializable/lib/src/json_part_builder.dart

```diff
@@ -23,10 +24,70 @@ Builder jsonPartBuilder({
 
   return SharedPartBuilder(
     [
-      JsonSerializableGenerator.fromSettings(settings),
+      _UnifiedGenerator([
+        JsonSerializableGenerator.fromSettings(settings),
+        const JsonEnumGenerator(),
+      ]),
       const JsonLiteralGenerator(),
     ],
     'json_serializable',
     formatOutput: formatOutput,
   );
 }
+
+/// Allows exposing separate [GeneratorForAnnotation] instances as one
```

Comment:
I don't understand why we need this.

What does this do differently than

```dart
SharedPartBuilder(
  [
    JsonSerializableGenerator.fromSettings(settonds),
     const JsonEnumGenerator(),
     const JsonLiteralGenerator(),
   ],
   'json_serializable',
   formatOutput: formatOutput,
 );
 ```

Reply:
It's weird, I know. I need them to be in the same generator so duplicates get de-duped.

Reply:
Can you expand the doc comment on the class to make that clear?

Reply:
https://github.com/google/json_serializable.dart/pull/990